src/line_detect/include/LineDetector.h

```diff
@@ -0,0 +1,133 @@
+/*
+ * Created By: Cameron Newton
+ * Created On: July 17, 2018
+ * Description: Class to scan cv::Mat object and return 3 parameter
+ * stuct representing orientation with pool marker
+ */
+
+#ifndef PROJECT_GATE_H
+#define PROJECT_GATE_H
+
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <unordered_map>
+
+/**
+*      Struct representing a marker on the bottom of the pool
+*
+*          width:   width of marker, used to interpolate distance from marker,
+* -1 if not seen
+*
+*          slope:   slope of marker seen, -1 if not seen
+*
+*          middleOfMarker:   vector representing middle of marker, -1 if not
+* seen
+*
+*          frontOfMarker:  distance to end of marker, -1 if not seen
+*
+*/
+struct LineToFollow {
+    float width;
+    float slope;
+    cv::Vec4i middleOfMarker;
+    int frontOfMarker;
+};
+
+/**
+*      Struct representing 2 markers seen on bottom of pool
+*
+*          frontLine: Marker to follow
+*
```

Comment:
we can get rid of some of this extra whitespace

Reply:
fixed